File: client/src/main/java/pro/panopticon/client/sensor/impl/ReportTimeSensor.java
package pro.panopticon.client.sensor.impl;

import pro.panopticon.client.model.Measurement;
import pro.panopticon.client.sensor.Sensor;

import java.time.LocalDateTime;
import java.util.List;

import static java.util.Collections.singletonList;

public class ReportTimeSensor implements Sensor {
    private static final String DESCRIPTION = "";
    @Override
    public List<Measurement> measure() {
        return singletonList(
                new Measurement("report.generated", "INFO", LocalDateTime.now().toString(), DESCRIPTION)
        );
    }
}

File: client/src/main/java/pro/panopticon/client/sensor/impl/ReportTimeSensor.kt
package pro.panopticon.client.sensor.impl

import pro.panopticon.client.model.Measurement
import pro.panopticon.client.sensor.Sensor
import java.time.LocalDateTime

class ReportTimeSensor : Sensor {
    override fun measure(): List<Measurement> {
        return listOf(
            Measurement(
                key = "report.generated",
                status = "INFO",
                displayValue = LocalDateTime.now().toString(),
                description = "",
            )
        )
    }
}
